### PR TITLE
[FIX] OWGeocoding: unmatched table editing

### LINKED ISSUE #38: GeoCoding: Match countries only on the output
##### Geo version

0.2.3

##### Orange version

3.14.0

##### Expected behavior
When setting identifiers in the Unmatched Identifiers section, identifiers are always the same as input data.

##### Actual behavior
When pressing commit, identifiers change internally. So if one makes a typo, the typo-ed version is now used as Unmatched identifier. This causes a mass of confusion.

##### Steps to reproduce the behavior
Use any geo data (e.g. HDI from Datasets). Set one of the Unmatched identifiers to X and press Commit. Now X is in Unmatched identifiers.

##### Additional info (worksheets, data, screenshots, ...)
screen shot 2018-07-02 at 14 52 26

### PR DESCRIPTION
**Pull request by robertcv**
##### Issue
Fixes #38
Additionally, the unmatched table was not user-friendly. 
##### Description of changes
Replace internal common name mapping to names that are in geojson data. Enable widget resizing so one can see longer names. Enable auto commit on edit finish of individual cells. Sort replacements to constrained their position in the table. Show the unmatched table even if there are no unmatched values to enable editing of replacements. 
##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/geo/cc_cities.py [hunk 1]
```diff
@@ -5,11 +5,10 @@
...
     'Kosovo': ['Kosovo, Republic of'],
-    'Northern Cyprus': ['N. Cyprus', 'North Cyprus'],
-    'Aland': ['Åland Islands', 'Åland'],
-    'Bosnia and Herzegovina': ['Bosnia and Herz.'],
-    'Czech Republic': ['Czech Rep.'],
-    'Faroe Islands': ['Faeroe Is.'],
+    'Åland': ['Åland Islands', 'Aland'],
+    'Bosnia and Herz.': ['Bosnia and Herzegovina'],
+    'Czechia': ['Czech Rep.', 'Czech Republic'],
+    'Faeroe Is.': ['Faroe Islands', 'Faeroe Islands'],
```
Comment (kernc) [inline orangecontrib/geo/cc_cities.py]: The mapping is [inverted later on](https://github.com/biolab/orange3-geo/pull/47/files#diff-e25d8f6b38b4b06194c1187a4380ea00R342). Left are the names used in GeoJSON files, right are synonyms. So some of these were in fact wrong?

Comment (kernc) [reply orangecontrib/geo/cc_cities.py]: For instance:
```bash
$ cd Orange3-geo/orangecontrib/geo/geojson
$ grep 'Lao PDF' *.json
admin0.json: ... "properties": {"name": "Lao PDR", "adm0_a3": "LAO", "_id": "LAO", "iso_a3": "LAO", "iso_a2": "LA", "sovereignt": "Laos", "type": "Sovereign country", "subregion": "South-Eastern Asia", "economy": "7. Least developed region", "fips10": "LA", "region_wb": "East Asia & Pacific", "continent": "Asia", "region_un": "Asia"}} ...

$ grep 'Czechia' *.json  # No match
$
```

Comment (robertcv) [reply orangecontrib/geo/cc_cities.py]: I am aware of the invert. I am working with the json files I got from the json branch. I infect get matches when I run `grep 'Czechia' *.json`. I change mappings were the key was not in CC_SHAPES (from the code I assume it contains the geojson data).

Comment (kernc) [reply orangecontrib/geo/cc_cities.py]: I regenerated the JSONs from sources and can no longer reproduce my above oservations. Your changes seem correct(er). :+1:

**Comment by robertcv** [comment]
@kernc Ajda will be absent for two weeks. He said I should assign someone else. Please finish the review and merge.

**Pull request merged**